sourcegraph/sourcegraph
# Code Insights: [FE] Connect capture group to the dashboard page

Follow up for https://github.com/sourcegraph/sourcegraph/pull/28817

Closes https://github.com/sourcegraph/sourcegraph/issues/28392 https://github.com/sourcegraph/sourcegraph/issues/28628

This PR is the first PR of three for finishing capture group creation UI
- https://github.com/sourcegraph/sourcegraph/pull/28817
- This one
- https://github.com/sourcegraph/sourcegraph/pull/28861

## Context

This PR extends the `Insight` model for the dashboard page with `CaptureGroupInsight`. This implies changes in
- GQL based and setting-based API (handling another type of insight)
- Changes in UI components 
- Reusing BackendView card for capture group insight (since it's fully compatible with SearchBased BE insight)

## How to test

Since we don't have any UI to connect the dashboard page and creation UI page we have to go to the creation UI capture group insight by URL. 

- Go to any dashboard
- Click on create insight button
- Go to Search based insight creation UI
- Replace in URL /search route to /capture-group
- Fill in the form
- Click create insight button 

In the end, you should see capture group insight attached to the dashboard that you came from. Colors of capture group insight should be different and based on our colors in the creation UI (same palette)
